kategory-core/src/test/kotlin/kategory/data/SortedMapKWTest.kt

```diff
@@ -0,0 +1,69 @@
+package kategory
+
+import io.kotlintest.KTestJUnitRunner
+import io.kotlintest.matchers.shouldNotBe
+import io.kotlintest.properties.forAll
+import org.junit.runner.RunWith
+
+@RunWith(KTestJUnitRunner::class)
+class SortedMapKWTest : UnitSpec() {
+
+    val EQ: Eq<HK2<SortedMapKWHK, String, Int>> = object : Eq<HK2<SortedMapKWHK, String, Int>> {
+        override fun eqv(a: HK2<SortedMapKWHK, String, Int>, b: HK2<SortedMapKWHK, String, Int>): Boolean =
+            a.ev().get("key") == b.ev().get("key")
+    }
+
+    val SG: Semigroup<Int> = object : Semigroup<Int> {
+        override fun combine(a: Int, b: Int): Int {
+            return a*b
+        }
+    }
+
+    init {
+
+        "instances can be resolved implicitly" {
+            functor<SortedMapKWHK>() shouldNotBe null
+            foldable<SortedMapKWHK>() shouldNotBe null
+            traverse<SortedMapKWHK>() shouldNotBe null
+            semigroup<SortedMapKWKind<String, Int>>() shouldNotBe null
+            monoid<SortedMapKWKind<String, Int>>() shouldNotBe null
+        }
+
+        val monoid = SortedMapKW.monoid<String, Int>(SG)
+
+        "Monoid Laws: identity" {
```

Comment:
This may be wrong in Map too. This should be tested using MonoidLaws.laws rather than in the file.

Reply:
@pakoito we only have MonoidkLaws :(

Reply:
We should then add Monoid and Semigroup laws before merging this. @Cotel wanna take care of those in this PR as well?

Reply:
I can try 💪